### ISSUE TITLE
PhenoTagger error & warning with BioBert model

### ISSUE THREAD

**Issue by m-vij** [ncbi-nlp/PhenoTagger#17]
Hi 

I am getting the following error and warning when using the following command: 

`python3 PhenoTagger_tagging.py -i ../example/input/ -o ../example/output/`

```
Some weights of the PyTorch model were not used when initializing the TF 2.0 model TFBertModel: ['cls.predictions.transform.LayerNorm.weight', 'cls.predictions.decoder.bias', 'cls.seq_relationship.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.decoder.weight', 'cls.predictions.bias', 'cls.predictions.transform.LayerNorm.bias', 'cls.seq_relationship.weight']
- This IS expected if you are initializing TFBertModel from a PyTorch model trained on another task or with another architecture (e.g. initializing a TFBertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing TFBertModel from a PyTorch model that you expect to be exactly identical (e.g. initializing a TFBertForSequenceClassification model from a BertForSequenceClassification model).
All the weights of TFBertModel were initialized from the PyTorch model.
If your task is similar to the task the model of the checkpoint was trained on, you can already use TFBertModel for predictions without further training.
Traceback (most recent call last):
  File "/Volumes/HardDrive/PHENO_HPO/PhenoTagger/src/PhenoTagger_tagging.py", line 280, in <module>
    phenotagger_tag(args.infolder,para_set,args.outfolder)
  File "/Volumes/HardDrive/PHENO_HPO/PhenoTagger/src/PhenoTagger_tagging.py", line 215, in phenotagger_tag
    nn_model=bioTag_BERT(vocabfiles)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/HardDrive/PHENO_HPO/PhenoTagger/src/nn_model.py", line 137, in __init__
    x = plm_model(x1_in, token_type_ids=x2_in, attention_mask=x3_in)[0][:,0,:]
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/HardDrive/PHENO_HPO/lib/python3.12/site-packages/tf_keras/src/utils/traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/Volumes/HardDrive/PHENO_HPO/lib/python3.12/site-packages/transformers/modeling_tf_utils.py", line 436, in run_call_with_unpacked_inputs
    unpacked_inputs = input_processing(func, config, **fn_args_and_kwargs)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/HardDrive/PHENO_HPO/lib/python3.12/site-packages/transformers/modeling_tf_utils.py", line 513, in input_processing
    raise ValueError(f"Data of type {type(v)} is not allowed only {allowed_types} is accepted for {k}.")
ValueError: Exception encountered when calling layer 'tf_bert_model' (type TFBertModel).

Data of type <class 'keras.src.backend.common.keras_tensor.KerasTensor'> is not allowed only (<class 'tensorflow.python.framework.tensor.Tensor'>, <class 'bool'>, <class 'int'>, <class 'transformers.utils.generic.ModelOutput'>, <class 'tuple'>, <class 'list'>, <class 'dict'>, <class 'numpy.ndarray'>) is accepted for token_type_ids.

Call arguments received by layer 'tf_bert_model' (type TFBertModel):
  • input_ids=<KerasTensor shape=(None, 32), dtype=int32, sparse=False, name=keras_tensor>
  • attention_mask=<KerasTensor shape=(None, 32), dtype=int32, sparse=False, name=keras_tensor_2>
  • token_type_ids=<KerasTensor shape=(None, 32), dtype=int32, sparse=False, name=keras_tensor_1>
  • position_ids=None
  • head_mask=None
  • inputs_embeds=None
  • encoder_hidden_states=None
  • encoder_attention_mask=None
  • past_key_values=None
  • use_cache=None
  • output_attentions=None
  • output_hidden_states=None
  • return_dict=None
  • training=False
```

I downloaded biobert-base-cased-v1.2 and biobert_PT_v1.2.h5 (renamed to biobert_PT.h5), and changed model_type to biobert in the para_set dictionary. 

```
models
└── biobert-base-cased-v1.2
    ├── biobert_PT.h5
    ├── config.json
    ├── gitattributes
    ├── pytorch_model.bin
    └── vocab.txt

```
Also, I had to download different versions of tensorflow, torch, and JPype1, due to pip being unable to install the ones mentioned in the requirements.txt. 

The issue seems to be about some incompatibility of keras and tensorflow software. Any guidance would be appreciated. 

Please let me know if you need any more information.

Thanks

**Comment by lingluodlut** [comment]
Yes. This issue seems to be about some incompatibility between new keras version and tensorflow. We only test our tool under tensorflow.keras mentioned in the requirements. What is the issue when you install the ones mentioned in the requirements.txt?

**Comment by m-vij** [comment]
Hi, thanks for your response

The issue with the requirements.txt version was that the latest python version (3.12.3) doesn't allow for installation of some of the mentioned library versions were not identifiable. For example, pip install tensorflow==2.12.0 gave

`ERROR: Could not find a version that satisfies the requirement tensorflow==2.12.0 (from versions: 2.16.0rc0, 2.16.1)
ERROR: No matching distribution found for tensorflow==2.12.0`. 

However, I was able to recently fix the issues by downgrading to python 3.9.

**Comment by lingluodlut** [comment]
Ok. Yes, the latest python can not install tensorflow 2.12.0.  Python 3.8-3.11 is OK.
